File: week2_Section_1.2/week2_1.2.py
var1 = "Global variable"

def outer_function():
    var2 = "I'm in outside function"

    def inner_function():
        var3 = "I'm in inside function"
        print(var3)  # Accessible here
        print(var2)  # Accessible due to closure
        print(var1) # Accessible (global scope)

    inner_function()
    #print(var3)  # Error: var3 is not accessible here

#print(var2) # Error: var2 is not accessible here
outer_function()

File: week2_Section_1.2/week2_1.2.cpp
// C++: Demonstrating Static Typing and Scope

#include <iostream>
using namespace std;

string var1 = "I'm global";

void outerFunction() {
    string var2 = "I'm in outside function";

    {
        string var3 = "I'm in inside block";
        cout << var3 << endl;  // Accessible
        cout << var2 << endl;  // Accessible
        cout << var1 << endl; // Accessible
    }

     //cout << var2 << endl;  // Error: var2 is not accessible here due to block scope
}

int main() {
    outerFunction();
    return 0;
}
